backend/mem/location.go

```diff
@@ -0,0 +1,211 @@
+package mem
+
+import (
+	"errors"
+	"github.com/c2fo/vfs/v4"
+	"github.com/c2fo/vfs/v4/utils"
+	"path"
+	"regexp"
+	"strings"
+	"time"
+)
+
+//Location implements the vfs.Location interface specific to in-memory fs.
+type Location struct {
+	exists     bool
+	firstTime  bool
+	Filename   string //the baseName of the file this location belongs to
+	name       string //the path that this location exists on
+	fileSystem vfs.FileSystem
+}
+
+//String implements io.Stringer by returning the location's URI as a string
+func (l *Location) String() string {
+
+	return l.URI()
+}
+
+/*
+List finds all of the files living at the current location and returns them in a slice of strings.
+If there are no files at location, then an empty slice will be returned
+*/
+func (l *Location) List() ([]string, error) {
+	//panic("implement me")
+
+	list := make([]string, 0)
+	str := l.Path()
+	for _, v := range fileList {
+		if v != nil {
```

Comment:
A bit of nitpicking here, but to reduce the number of indentation (which is the True Go Style), you can do 

``` go
for _, v := range fileList {
  if v == nil {
   continue
  }
  // my stuff but without an additional indent.
```